Aphidius ervi EE transcriptomics
================================

- By Mark Lammers ©, marklammers@uni-muenster.de
- Repo started: 01.02.2023 ('%d.%m.%Y')
- Last update:  01.02.2023

# A flexibly fluctuating but targeted todo-list

- [ ] Data handling
  - [x] Verify data integrity
  - [x] Inspect MultiQC output
  - [x] Make new FastQC and MultiQC reports
  - [x] Import sample information and match with file names
  - [ ] Make separate MultiQC reports per tissue type
  - [ ] Generate new multiqc results per tissue type
- [ ] Data processing
  - [ ] Choose software for adapter trimming
  - [ ] Run adapter trimming on all samples
  - [ ] Choose software for mapping reads to the reference genome
  - [ ] Copy latest version of the reference genome into 0.data
  - [ ] Benchmark mapping to reference genome on `jgant3`
  - [ ] Map all reads to the reference genome
- [ ] DGE analysis
  - [ ] Choose pipeline
  - [ ] Calculate FPKM


# Project Setup

See `1.code/project.setup.md`

# Sample information

- Species: Aphidius ervi strain 'Katz'
- Experiment: Multi-generational tissue-specific transcriptomic differences due to host adaptation.
- Input material: 156 RNA extracts of varying quality from insect tissue samples.
- Tissues: heads, ovaries, venom glands.
- Method: Lexogen QuantSeq 3′ mRNA-Seq (Illumina), see `https://www.lexogen.com/docs/quantseq/`.
- Data: Illumina RNAseq, 312+19 libraries.

See also `0.data/mlammers_ML01_*/Sample_Names.tab` for specific samples names.

This information is also collated in `0.data/Sample.info.tab` and `0.data/Files.info.tab` after running `1.code/collect.sample.info.R`.

# Handling and assessing raw data

## Retrieve data

See `0.data/instructions_big_data_set.txt`
and `0.data/instructions_small_data_set.txt`.

## Check data integrity

md5sums generated by the Cologne sequenceing center, before retrieval of data, are stored in `0.data/mlammers_ML01_*/md5_checksums.tab`.

Combine them into one file:
```bash
cat 0.data/mlammers_ML01_Juni1/md5_checksums.tab >0.data/md5sums_original.txt
cat 0.data/mlammers_ML01_Mai11/md5_checksums.tab >>0.data/md5sums_original.txt
```

Generate local md5sums:
```bash
md5sum 0.data/mlammers_ML01_*/A00* >>0.data/md5sums_local.txt
sed -i 's/0.data\/mlammers_ML01_Juni1\///g' 0.data/md5sums_local.txt
sed -i 's/0.data\/mlammers_ML01_Mai11\///g' 0.data/md5sums_local.txt
sed -i 's/  / /g' 0.data/md5sums_local.txt
```
Compare files with checksums:
```bash
md5sum 0.data/md5sums_*.txt
# 495a242412612329a2f7d42b54c792cf  0.data/md5sums_local.txt
# 495a242412612329a2f7d42b54c792cf  0.data/md5sums_original.txt
# So the checksums stored in both files, as well as the file names, must be identical between these files
```

## Store a backup of the data on the IEB server
```bash
mkdir /global/scratch2/mlammer1/aervi-ee-rnaseq
rsync -ah --bwlimit=1500 --info=progress2 ./0.data/* /global/scratch2/mlammer1/aervi-ee-rnaseq/
# Expected runtime at 1.5 MB/s is about 17.5 hours
```

## Quality control of raw data

The sequencing center provided `multiqc` output with the two data sets.

See `0.data/mlammers_ML01_*/FastQC/multiqc_report.html`.
Copy the files to `3.results`, add to repo, and inspect:
```bash
cp -v 0.data/mlammers_ML01_Mai11/FastQC/multiqc_report.html 3.results/multiqc_report_Mai11.html
cp -v 0.data/mlammers_ML01_Juni1/FastQC/multiqc_report.html 3.results/multiqc_report_Juni1.html
git add -f 3.results/multiqc_report_Mai11.html 3.results/multiqc_report_Juni1.html
```

Some conclusions on sequence data quality:
- Juni1:
  - 19 samples
  - %Dups range 36.6-77.5% -> also reflected in plots of overrepresented sequences
  - $GC range 32-42%
  - Read length always 101bp
  - 4.2-9.1M reads per sample
  - Sequence quality histograms look excellent
  - Up to 29% of sequences have adapter content towards the end of the read
- Mai11:
  - 312 samples
  - %Dups range 34.8-92.5% -> also reflected in plots of overrepresented sequences
  - %GC range 32-48%
  - Read length always 101bp
  - 2.8-10.7M reads per sample
  - Sequence quality histograms look excellent
  - Up to 36% of sequences have adapter content towards the end of the read

Quite a spread in certain variables. Two tasks added.

### Compare data quality per tissue type

```bash
# Do new FastQC analysis so that I also have the .zip output
#Mai11
mkdir 0.data/mlammers_ML01_Mai11/own-fastqc
fastqc --outdir 0.data/mlammers_ML01_Mai11/own-fastqc \
  --threads 16 0.data/mlammers_ML01_Mai11/A00*.fastq.gz
#Juni1
mkdir 0.data/mlammers_ML01_Juni1/own-fastqc
fastqc --outdir 0.data/mlammers_ML01_Juni1/own-fastqc \
  --threads 16 0.data/mlammers_ML01_Juni1/A00*.fastq.gz

# Run multiqc on all files together
pip install --user multiqc
ls -lha 0.data/mlammers_ML01_*/own-fastqc/A00*_fastqc.zip | less
mkdir 3.results/own_multiqc_reports
python3 -m multiqc --interactive -o 3.results/own_multiqc_reports \
  0.data/mlammers_ML01_*/own-fastqc/A00*_fastqc.zip

# Run multiqc per tissue type
#TODO
```

# Data processing

## Adapter trimming

(if necessary)

## Mapping to reference genome

TODO
