Aphidius ervi EE transcriptomics
================================

- By Mark Lammers ©, marklammers@uni-muenster.de
- Repo started: 01.02.2023 ('%d.%m.%Y')
- Last update:  01.02.2023

# Project Setup

See `1.code/project.setup.md`

# Sample information

- Species: Aphidius ervi 'Katz'
- Experiment: Multi-generational tissue-specific transcriptomic differences due to host adaptation.
- Input material: 156 RNA extracts of varying quality from insect tissue samples.
- Tissues: heads, ovaries, venom glands.
- Method: Lexogen QuantSeq 3′ mRNA-Seq (Illumina), see `https://www.lexogen.com/docs/quantseq/`.
- Data: Illumina RNAseq.

See also `0.data/mlammers_ML01_*/Sample_Names.tab` for specific samples names.

# Handling and assessing raw data

## Retrieve data

See `0.data/instructions_big_data_set.txt`
and `0.data/instructions_small_data_set.txt`.

## Check data integrity

md5sums generated by the Cologne sequenceing center, before retrieval of data, are stored in `0.data/mlammers_ML01_*/md5_checksums.tab`.

Combine them into one file:
```bash
cat 0.data/mlammers_ML01_Juni1/md5_checksums.tab >0.data/md5sums_original.txt
cat 0.data/mlammers_ML01_Mai11/md5_checksums.tab >>0.data/md5sums_original.txt
```

Generate local md5sums:
```bash
md5sum 0.data/mlammers_ML01_*/A00* >>0.data/md5sums_local.txt
sed -i 's/0.data\/mlammers_ML01_Juni1\///g' 0.data/md5sums_local.txt
sed -i 's/0.data\/mlammers_ML01_Mai11\///g' 0.data/md5sums_local.txt
sed -i 's/  / /g' 0.data/md5sums_local.txt
```
Compare files with checksums:
```bash
md5sum 0.data/md5sums_*.txt
# 495a242412612329a2f7d42b54c792cf  0.data/md5sums_local.txt
# 495a242412612329a2f7d42b54c792cf  0.data/md5sums_original.txt
# So the checksums stored in both files, as well as the file names, must be identical between these files
```

## Store a backup of the data on the IEB server
```bash
mkdir /global/scratch2/mlammer1/aervi-ee-rnaseq
rsync -ah --bwlimit=1500 --info=progress2 ./0.data/* /global/scratch2/mlammer1/aervi-ee-rnaseq/
# Expected runtime at 1.5 MB/s is about 17.5 hours
```

## Quality control of raw data

The sequencing center provided `multiqc` output with the two data sets.

See `0.data/mlammers_ML01_*/FastQC/multiqc_report.html`

# Data processing

## Adapter trimming

(if necessary)

## Mapping to reference genome

TODO
